packages/app-utils/src/components/ConnectionBootstrap.tsx

```diff
@@ -104,21 +186,43 @@ export function ConnectionBootstrap({
     [objectFetcher]
   );
 
-  if (connection == null || error != null) {
+  function handleRefresh(): void {
+    log.info('Refreshing application');
+    window.location.reload();
+  }
+
+  if (isShutdown) {
```

Comment:
This should also show if the status is `connecting` (i.e. the initial connection). Previously that's what was shown here

Also, there is nothing that sets the status to `connected` on the first connection. Add that where `setConnection` is called so you can then do `isShutdown || connectionStatus === 'connecting'`. Previously this was `if connection == null || error != null` which was initial connection or shutdown. The reconnection and auth failed were handled in the app.